File: src/chisel/checker/models/project_info.py
from __future__ import annotations

from dataclasses import dataclass, field
from pathlib import Path

from chisel.checker.models.file_info import FileInfo


@dataclass(slots=True)
class ProjectInfo:
    root_path: Path
    files: list[FileInfo] = field(default_factory=list)
    package_name: str = ""

File: chisel_js/src/chisel/checker/models/project_info.ts
import type { FileInfo } from "./file_info";

export interface ProjectInfo {
  readonly rootPath: string;
  readonly files: readonly FileInfo[];
  readonly packageName: string;
}

export function createProjectInfo(params: {
  rootPath: string;
  files?: FileInfo[];
  packageName?: string;
}): ProjectInfo {
  return Object.freeze({
    rootPath: params.rootPath,
    files: Object.freeze(params.files ?? []),
    packageName: params.packageName ?? "",
  });
}
